consensus/reactor.go

```diff
@@ -1163,8 +1186,8 @@ func (ps *PeerState) SetHasProposalBlockPart(height int64, round int32, index in
 }
 
 // PickSendVote picks a vote and sends it to the peer.
-// Returns true if vote was sent.
-func (ps *PeerState) PickSendVote(votes types.VoteSetReader) bool {
+// Returns the vote if vote was sent. Otherwise, returns nil.
+func (ps *PeerState) PickSendVote(votes types.VoteSetReader) *types.Vote {
```

Comment:
since we want to trace each vote, wdyt of just adding the write in this function? this way we ensure that we are not missing any sending of votes and have a single place where we call WriteVote

Reply:
Initially, I had the same idea, but realized that within this function, we don't have access to the sending peer's state, namely their height and round. This led me to shift the tracing to the caller function instead.

Reply:
ahh I see. do you think it would be worth passing the round state?

Reply:
The inclusion of the round state in the `PickSendVote` logic is primarily for information tracing from the caller's side and does not directly relate to the core logic. While this approach may not seem the cleanest, I will not object to this change. However,  if we do so, further refactoring will be needed: either passing the trace client of the consensus reactor to the function or adding a trace client field to the `PeerState`.